my-app/src/Utilities/ValidationUtilities.tsx

```diff
@@ -123,6 +125,42 @@ export function isValidMove(
                .map(point => getPieceAtLocation(point))
                   .some(piece => piece !== null);      
    }
+
+   /*  
+      castling is only possible if neither the king nor the rook has moved
+      there must not be any pieces between the king and the rook
+      the king may not be in check
+      the square the king goes to and any intervening squares may not be under attack
+      */
+   function pieceIsCastlingAndCannotCastle(moveLocation: Coordinate, piece: Piece, board: Array<Array<(Piece | null)>>) : boolean {
+      if (piece.pieceName !== Pieces.KING) return false;
+      if (!(piece as King).isCastlingMove(moveLocation)) return false;
+
+      let correspondingCastle: Piece | null = getCorrespondingCastle(moveLocation, board);
```

Comment:
Should this be named getCorrespondingRook? I always get confused what the official name for the piece is, but the piece name is Rook in here so seems like it should be the same.

Reply:
woops, yeah Rook makes more sense